microservices/api/src/controllers/roles/Update.php

```diff
@@ -34,9 +34,10 @@ public function run( int $id): array  {
           $role->last_modified_by = $commonFields['last_modified_by'];
           $role->last_modified_date = $commonFields['last_modified_date'];
         }
-        if (strtolower($role->scope) <> 'project'){
-          $role->project_id = null;
+        if (strtolower($role->scope) <> 'project' || strtolower($role->type) <> 'custom') {
```

Comment:
From my understanding:

In the original logic, we nullified the project_id only if the scope wasn't 'project'.
In the updated logic, we're nullifying the project_id if the scope isn't 'project' OR if the type isn't 'custom'.

Can you provide insights on:
The specific requirements or issues that prompted this change?
Whether there are roles with a 'project' scope and a type other than 'custom' that we still want to retain their project_id?
Any additional context behind this decision that would help clarify the updated behavior?
I believe this will help us ensure the change aligns with our expectations and will also assist in future reference.

Thanks in advance!

Reply:
For me 

> guys, what do you think about the comparison operator "<>"? for me the classical one "!=" looks more readable

For me it's better to use != and !== for different than and not exactly equal to. I'll change this.